src/Numerics/Mesh/Grids.jl

```diff
@@ -570,106 +591,242 @@ function commmapping(N, commelems, commfaces, nabrtocomm)
 end
 
 # {{{ compute geometry
-function computegeometry(
-    topology::AbstractTopology{dim},
-    D,
-    ξ,
-    ω,
-    meshwarp,
-    vmap⁻,
-) where {dim}
-    # Compute metric terms
+function computegeometry_fvm(elemtocoord, D, ξ, ω, meshwarp)
+    FT = eltype(D[1])
+    dim = length(D)
+    nface = 2dim
+    nelem = size(elemtocoord, 3)
+
     Nq = ntuple(j -> size(D[j], 1), dim)
     Np = prod(Nq)
     Nfp = div.(Np, Nq)
 
-    FT = eltype(D[1])
+    Nq_N1 = max.(Nq, 2)
+    Np_N1 = prod(Nq_N1)
+    Nfp_N1 = div.(Np_N1, Nq_N1)
+
+    # First we compute the geometry with all the N = 0 dimension set to N = 1
+    ξ1, ω1 = Elements.lglpoints(FT, 1)
+    D1 = Elements.spectralderivative(ξ1)
+    D_N1 = ntuple(j -> Nq[j] == 1 ? D1 : D[j], dim)
+    ξ_N1 = ntuple(j -> Nq[j] == 1 ? ξ1 : ξ[j], dim)
+    ω_N1 = ntuple(j -> Nq[j] == 1 ? ω1 : ω[j], dim)
+    (vgeo_N1, sgeo_N1, x_vtk) =
+        computegeometry(elemtocoord, D_N1, ξ_N1, ω_N1, meshwarp)
+
+
+    # Sort out the vgeo terms
+    @views begin
+        vgeo_N1_flds =
+            ntuple(fld -> reshape(vgeo_N1[:, fld, :], Nq_N1..., nelem), _nvgeo)
+
+        # Allocate the storage for N = 0 volume metrics
+        vgeo = zeros(FT, Np, _nvgeo, nelem)
+
+        # Counter to make sure we got all the vgeo terms
+        num_vgeo_handled = 0
+
+        # _M should be a sum
+        vgeo[:, _M, :][:] .= sum(vgeo_N1_flds[_M], dims = findall(Nq .== 1))[:]
+        num_vgeo_handled += 1
+
+        # need to recompute _MI
+        vgeo[:, _MI, :] = 1 ./ vgeo[:, _M, :]
+        num_vgeo_handled += 1
+
+        # coordinates should just be averages
+        avg_den = 2 .^ sum(Nq .== 1)
+        for fld in (_x1, _x2, _x3)
+            vgeo[:, fld, :] =
+                sum(vgeo_N1_flds[fld], dims = findall(Nq .== 1))[:] / avg_den
+            num_vgeo_handled += 1
+        end
+
+        # For the metrics it is J * ξixk we approximate so multiply and divide the
+        # mass matrix (which has the Jacobian determinant and the proper averaging
+        # due to the quadrature weights)
+        M_N1 = vgeo_N1_flds[_M]
+        MI = vgeo[:, _MI, :]
+        for fld in
+            (_ξ1x1, _ξ2x1, _ξ3x1, _ξ1x2, _ξ2x2, _ξ3x2, _ξ1x3, _ξ2x3, _ξ3x3)
+            vgeo[:, fld, :] =
+                sum(M_N1 .* vgeo_N1_flds[fld], dims = findall(Nq .== 1))[:] .*
+                MI[:]
+            num_vgeo_handled += 1
+        end
+
+        # compute MH and JvC
+        horizontal_vertical_metrics(vgeo, Nq, ω)
+        num_vgeo_handled += 2
+
+        # Make sure we handled all the vgeo terms
+        @assert _nvgeo == num_vgeo_handled
+    end
+
+    # Sort out the sgeo terms
+    @views begin
+        sgeo = zeros(FT, _nsgeo, maximum(Nfp), nface, nelem)
+
```

Comment:
Is Nfp the number of points in each face?

Reply:
Yes.